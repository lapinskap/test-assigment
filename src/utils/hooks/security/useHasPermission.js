import { useEffect } from 'react';
import RbsContext from '../../RoleBasedSecurity/RbsContext';
import { godPermission } from '../../RoleBasedSecurity/permissions';

const { useState, useContext } = require('react');

const useHasPermission = (aclKey) => {
  const { userInfo } = useContext(RbsContext);
  const [result, setResult] = useState(null);
  useEffect(() => {
    // const permissions = userInfo.getPermissions();
    // setResult(Boolean(permissions.includes(godPermission) || permissions.includes(aclKey)));
    setResult(true);
  }, [userInfo, aclKey]);

  return result;
};

export default useHasPermission;
