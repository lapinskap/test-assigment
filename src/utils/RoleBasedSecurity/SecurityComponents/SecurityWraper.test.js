import React from 'react';
import renderer from 'react-test-renderer';
import { getPermissionsWrapper } from '../../hooks/security/useHasPermission.test';
import SecurityWrapper from './SecuirityWrapper';

describe('<RbsButton />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when have permissions ', () => {
    const Wrapper = getPermissionsWrapper(['test1:company:company']);
    const tree = renderer
      .create(
        <Wrapper>
          <SecurityWrapper
            permission="test1:company:company"
          >
            <div>content</div>
          </SecurityWrapper>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders nothing when have not permissions ', () => {
    const Wrapper = getPermissionsWrapper(['test1:company:company']);

    const tree = renderer
      .create(
        <Wrapper>
          <SecurityWrapper
            permission="random:scope"
          >
            <div>content</div>
          </SecurityWrapper>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders disabled content nothing when have not permissions and disable props', () => {
    const Wrapper = getPermissionsWrapper(['test1:company:company']);

    const tree = renderer
      .create(
        <Wrapper>
          <SecurityWrapper
            disable
            permission="random:scope"
          >
            <div>content</div>
          </SecurityWrapper>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
