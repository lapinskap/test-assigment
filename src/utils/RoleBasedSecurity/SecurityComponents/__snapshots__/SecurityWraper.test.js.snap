// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RbsButton /> renders correctly when have permissions  1`] = `null`;

exports[`<RbsButton /> renders disabled content nothing when have not permissions and disable props 1`] = `
<div
  className="disabled-area-wrapper"
  title="Nie masz uprawnieÅ„ (random:scope)"
>
  <div
    className="not-allowed disabled-area"
  />
  <div>
    content
  </div>
</div>
`;

exports[`<RbsButton /> renders nothing when have not permissions  1`] = `null`;
